src/shared-components/tooltip/index.tsx

```diff
@@ -0,0 +1,139 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { Global } from '@emotion/core';
+import Arrow from 'src/svgs/icons/arrow.svg';
+
+import OffClickWrapper from '../offClickWrapper';
+import {
+  MainContainer,
+  Trigger,
+  TooltipBox,
+  TooltipContent,
+  ArrowImageContainer,
+} from './style';
+import { COLORS } from '../../constants';
+
+export type ArrowAlignTypes = 'left' | 'middle' | 'right';
+
+export type PositionTypes = 'top' | 'bottom';
+
+type TooltipProps = {
+  alignRightPercent?: number;
+  alignTopPercent?: number;
+  arrowAlign?: ArrowAlignTypes;
+  children: React.ReactNode;
+  content?: React.ReactNode;
+  defaultOpen?: boolean;
+  display?: boolean;
+  hasRestrictedWidth?: boolean;
+  isSmall?: boolean;
+  nudgeLeft?: number;
+  nudgeRight?: number;
+  nudgeTop?: number;
+  nudgeBottom?: number;
+  position?: PositionTypes;
+};
+
+const Tooltip = ({
+  alignRightPercent = 0,
+  alignTopPercent = 0,
+  arrowAlign = 'middle',
+  children,
+  content = '',
+  defaultOpen = false,
+  display = true,
+  hasRestrictedWidth = false,
+  isSmall = false,
+  nudgeLeft = 0,
+  nudgeRight = 0,
+  nudgeTop = 0,
+  nudgeBottom = 0,
+  position = 'top',
+}: TooltipProps) => {
+  const [clicked, setClicked] = useState(false);
+  const [hovered, setHovered] = useState(false);
+
+  const onClick = () => {
+    if (clicked) {
+      // if clicked is true, we're about to update to false so remove class
+      document?.querySelector('body')?.classList.remove('cursor-pointer');
+    } else {
+      document?.querySelector('body')?.classList.add('cursor-pointer');
+    }
+
+    setClicked(!clicked);
+  };
+
+  const closeTooltip = () => {
+    if (clicked) {
+      document?.querySelector('body')?.classList.remove('cursor-pointer');
+    }
+
+    setClicked(false);
+    setHovered(false);
+  };
+
+  const open = defaultOpen || clicked || hovered;
+
+  return (
+    <OffClickWrapper onOffClick={closeTooltip}>
+      <Global
+        styles={{
+          'body.cursor-pointer': {
+            cursor: 'pointer',
+          },
```

Comment:
i never knew we added cursor pointer to the whole body 🤔

Reply:
yeah this is old logic, what happens is that if u click the trigger element the tooltip stays opened until u click somewhere else so the cursor pointer maybe symbolise that UX that u need click somewhere in the page for the tooltip to go away. Also I think when I reworked the tooltip we were just adding the class and relying on the app to add that class, I think I added the GlobalStyle here so it decouple this logic from the app